mne_bids/write.py

```diff
@@ -1249,37 +1308,66 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
                           manufacturer, coordsystem_fname, kind,
                           overwrite, verbose)
-    elif kind == 'ieeg':
-        coord_frame = "mri"  # defaults to MRI coordinates
-        unit = "m"  # defaults to meters
-    else:
-        coord_frame = None
-        unit = None
 
+    # write electrodes data for iEEG and EEG
+    electrodes_fname = make_bids_basename(
+        subject=subject_id, session=session_id, acquisition=acquisition,
+        suffix='electrodes.tsv', prefix=data_path)
+    unit = "m"  # defaults to meters
     # We only write electrodes.tsv and accompanying coordsystem.json
     # if we have an available DigMontage
     if raw.info['dig'] is not None:
         if kind == "ieeg":
+            # get coordinate frame from digMontage
+            digpoint = raw.info['dig'][0]
+            if any(digpoint['coord_frame'] != _digpoint['coord_frame']
+                   for _digpoint in raw.info['dig']):
+                warn("Defaulting coordinate frame of iEEG coords to "
+                     "'mri' since not all digpoints "
+                     "have the same coordinate frame.")
+                coord_frame = "mri"  # defaults to MRI coordinates
+            else:
+                # get the accepted mne-python coordinate frames
+                coord_frame_int = int(digpoint['coord_frame'])
+                coord_frame = _verbose_ieeg_coordinate_frames.\
+                    get(coord_frame_int, None)
+
+                # default coordinate frame to mri if not available
+                if coord_frame is None:
+                    coord_frame = 'mri'
+
             coords = _extract_landmarks(raw.info['dig'])
 
-            # Rescale to MNE-Python "head" coord system, which is the
-            # "ElektaNeuromag" system (equivalent to "CapTrak" system)
+            # RPA, NAS and LPA shouldn't be set in iEEG
             if set(['RPA', 'NAS', 'LPA']) != set(list(coords.keys())):
                 # Now write the data to the elec coords and the coordsystem
                 _electrodes_tsv(raw, electrodes_fname,
                                 kind, overwrite, verbose)
                 _coordsystem_json(raw, unit, orient,
                                   coord_frame, coordsystem_fname, kind,
                                   overwrite, verbose)
-        elif kind != "meg":
+        elif kind == 'eeg':
+            # We only write EEG electrodes.tsv and coordsystem.json
+            # if we have LPA, RPA, and NAS available to rescale to a known
+            # coordinate system frame
+            coords = _extract_landmarks(raw.info['dig'])
+            if set(['RPA', 'NAS', 'LPA']) != set(list(coords.keys())):
+                warn("Writing of EEG electrodes.tsv is skipped here... "
+                     "It only occurs if "
+                     "we have 'RPA', 'NAS', and 'LPA'. Please "
+                     "set these and refer to MNE-Python for more info.")
```

Comment:
Again: Can we use `logger` to emit a warning?

Would recommend this string: `Writing EEG electrodes.tsv is only possible if anatomical landmarks (NAS, LPA, RPA) are present. Skipping ...`